pkg/operator/k8sutil/resources.go

```diff
@@ -24,8 +24,55 @@ import (
 	"github.com/ghodss/yaml"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
+// OwnerInfo is to set owner references. Only one of owner and ref must be valid.
+type OwnerInfo struct {
+	owner    metav1.Object
+	scheme   *runtime.Scheme
+	ownerRef *metav1.OwnerReference
+}
+
+// NewOwnerInfo create a new ownerInfo to set ownerReference by controllerutil
+func NewOwnerInfo(owner metav1.Object, scheme *runtime.Scheme) *OwnerInfo {
+	return &OwnerInfo{owner: owner, scheme: scheme}
+}
+
+// NewOwnerInfoWithOwnerRef create a new ownerInfo to set ownerReference by rook itself
+func NewOwnerInfoWithOwnerRef(ownerRef *metav1.OwnerReference) *OwnerInfo {
+	return &OwnerInfo{ownerRef: ownerRef}
+}
+
+// SetControllerReference set the controller reference of object
+func (info *OwnerInfo) SetControllerReference(object metav1.Object) error {
+	if info.owner != nil {
+		return controllerutil.SetControllerReference(info.owner, object, info.scheme)
+	} else {
```

Comment:
nit: no need for the `else` block since the `if` block returned.

Reply:
fixed, thanks.